network/hostnetwork/dht.go

```diff
@@ -901,18 +904,23 @@ func (dht *DHT) dispatchPacketType(ctx hosthandler.Context, msg *packet.Packet,
 	packetBuilder := packet.NewBuilder().Sender(ht.Origin).Receiver(msg.Sender).Type(msg.Type)
 
 	// TODO: fix sign and check sign logic
-	// if msg.Type == packet.TypeRPC {
-	// 	data := msg.Data.(*packet.RequestDataRPC)
-	// 	signedMsg, err := message.Deserialize(bytes.NewBuffer(data.Args[0]))
-	// 	if err != nil {
-	// 		log.Error(err, "failed to parse incoming RPC")
-	// 		return
-	// 	}
-	// 	if !message.SignIsCorrect(signedMsg, dht.GetNetworkCommonFacade().GetSignHandler().GetPrivateKey()) {
-	// 		log.Warn("RPC message not signed")
-	// 		return
-	// 	}
-	// }
+	if msg.Type == packet.TypeRPC {
+		data := msg.Data.(*packet.RequestDataRPC)
+		signedMsg, err := message.Deserialize(bytes.NewBuffer(data.Args[0]))
+		if err != nil {
+			log.Error(err, "failed to parse incoming RPC")
+			return
+		}
+		activeNode := dht.activeNodeKeeper.GetActiveNode(*data.NodeID)
```

Comment:
refactor lines 914-921 in a function (smth like `checkActiveNodeSign(keeper NodeKeeper, msg *packet.RequestDataRPC)`) and write unit tests for it